### PR TITLE
Feat: Integrate the POST call to QR auth

### LINKED ISSUE #153: Integrate the POST call
Integrate POST api call and send the scanned user-id to the backend

**Dependency**
- [x] https://github.com/Real-Dev-Squad/website-backend/issues/941

### PR DESCRIPTION
**Pull request by biswanath-tewari**
### Closes: [153](https://github.com/Real-Dev-Squad/mobile-app/issues/153)

Integrates POST api call and sends the scanned user-id to the backend
NOTE: scanner implementation is yet to be done, have used a placeholder for the user-id

### CODE REVIEW

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 1]
```diff
@@ -59,6 +61,51 @@ const AuthScreen = () => {
...
+        },
+        body: JSON.stringify({
+          device_info: deviceInfo,
+          user_id: 'BE9a4sGXFLDwxZU3DSiq', //TODO: replace with scanner results
+          device_id: deviceId,
+        }),
+      });
+
+      if (data.ok) {
+        console.log('patch call successfull');
```
Comment (sahsisunny) [inline src/screens/AuthScreen/AuthScreen.tsx:83]: Why log?

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 2]
```diff
@@ -59,6 +61,51 @@ const AuthScreen = () => {
...
+      } else {
+        Toast.show({
+          type: 'error',
+          text1: 'Somethin went wrong, please try again',
+          position: 'bottom',
+          bottomOffset: 80,
+        });
+      }
+    } catch (err) {
+      console.log(err);
```
Comment (sahsisunny) [inline src/screens/AuthScreen/AuthScreen.tsx]:
```suggestion
      console.error(err);
```

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 1 above)
Comment (biswanath-tewari) [reply src/screens/AuthScreen/AuthScreen.tsx:83]: it is placeholder for the actual functionality

File: src/constants/apiConstant/AuthApi.ts
Comment (sahsisunny) [inline src/constants/apiConstant/AuthApi.ts:1]: Why you created a separate file for this one URL?

Comment (biswanath-tewari) [reply src/constants/apiConstant/AuthApi.ts:1]: following the convention with other implementation for consistency

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 2 above)
Comment (biswanath-tewari) [reply src/screens/AuthScreen/AuthScreen.tsx]: done

File: src/constants/apiConstant/AuthApi.ts
Comment (biswanath-tewari) [reply src/constants/apiConstant/AuthApi.ts:1]: as per our discussion, we decided to goi ahead with the AuthApi file itself

**Review by shreya-mishra — COMMENTED** [review]
will  you please remove this yarn.lock file

**Review by shreya-mishra — APPROVED** [review]

**Pull request merged**